django/db/models/fields/__init__.py

```diff
@@ -738,7 +738,8 @@ def db_returning(self):
         Private API intended only to be used by Django itself. Currently only
         the PostgreSQL backend supports returning multiple fields on a model.
         """
-        return False
+        from django.db.models.expressions import BaseExpression
+        return self.has_default() and isinstance(self.default, BaseExpression)
```

Comment:
We usually use duck typing for such cases

```suggestion
        return self.has_default() and hasattr(self.default, 'as_sql')
```